plugin/src/main/resources/seedJobGroovyScript.groovy

```diff
@@ -119,28 +118,22 @@ def updateJobs(FilePath filePath, String namePrefix = '') {
 def workspace = build.workspace
 def xmlBase = new FilePath(workspace, 'XML_BASE_DIR_PLACEHOLDER')
 
-def localFile = new File("TEMPORARY_XML_DIR_PREFIX_PLACEHOLDER-${build.getNumber()}")
-println "Create local XML dir '${localFile.absolutePath}'"
-if (localFile.exists()) {
-    throw new RuntimeException("Local XML dir '${localFile.absolutePath}' already exists")
-}
-if (!localFile.mkdir()) {
-    throw new RuntimeException("Could not create local XML dir '${localFile.absolutePath}'")
-}
-def localXmlBase = new FilePath(localFile)
+def localDir = Files.createTempDirectory("TEMPORARY_XML_DIR_PREFIX_PLACEHOLDER")
+println "Created local XML dir '${localDir}'"
+def localXmlBase = new FilePath(new File(localDir.toString()))
 
 println 'Copy remote files to local XML dir'
 xmlBase.copyRecursiveTo(localXmlBase)
 
 println 'Start updating jobs from local XML dir'
 updateJobs(localXmlBase)
+println 'Updating jobs done'
 
-println "Delete local XML dir '${localFile.absolutePath}'"
-if (!localFile.deleteDir()) {
-    throw new RuntimeException("Could not delete local XML dir '${localFile.absolutePath}'")
-}
+println "Deleting local XML dir '${localDir}'"
+Files.delete(localDir)
```

Comment:
This will throw a `DirectoryNotEmptyException`:
https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#delete(java.nio.file.Path)

I agree that throwing an exception is too much here, but we should at least print an error message:

```
if (!localDir.toFile().deleteDir()) {
  println "Could not delete local XML dir '${localDir}'."
}
```

Reply:
Done